packages/editor/src/components/rich-text/index.js

```diff
@@ -780,6 +737,18 @@ export class RichText extends Component {
 			record.end = length;
 			this.applyRecord( record );
 		}
+
+		if ( isSelected && ! prevProps.isSelected ) {
```

Comment:
@aduth Better?

Reply:
We still have a condition in `onSelectionChange` which tests whether the rich text is the active one, which sends mixed messaging as to whether the intent of the application of event handlers _here_ is meant to ensure they're only bound if the field is active/selected.

Reply:
Furthermore, it's unclear what distinction is meant to be made between `this.isActive()` and `this.props.isSelected`, which seems important to clarify as we're making assumptions about `isSelected` as a basis for assuming an active RichText.

Reply:
Looking through how `isSelected` becomes assigned, I don't know that it's safe currently to make the assumption that `isSelected` can imply the field is focused. We only call `setFocusedElement` on `focus`, but never anything when focus leaves the field. While only one RichText can be the focused element at a given time, I still think it's possible for a `RichText` to be the "focused instance" despite the _actual DOM focus_ being elsewhere in the page (title, for example). This feels like it should be improved, and blocks being able to use `isSelected` here.

Ideally, I think we consolidate `isActive` with `isSelected` such that `isSelected` can only be true if and only if the `RichText` field is focused.

Reply:
I think [the comment](https://github.com/WordPress/gutenberg/pull/6871#issuecomment-392895193) from @youknowriad is the best way to describe why `blur` is very hard to implement in this case:

> The issue is that sometimes RichText loses the focus but we still want to keep it selected. (Link modal, toolbar buttons), It's probably doable but not simple because of the fact that popovers show outside the RichText's node.

This issue has a very long history. Initially focus had to be manually maintained using `isSelected` and `onFocus` prop. `onFocus` was proxied through `RichText` to `TinyMCE` but it has changed in #6286. I tried to get rid of their usage in core with #6871 but we decided that it isn't ideal because it doesn't handle `blur` event at all. To move forward we opened #7029 which renamed `onFocus` to `unstableOnFocus`.

> Ideally, I think we consolidate `isActive` with `isSelected` such that `isSelected` can only be true if and only if the `RichText` field is focused.

I'm not quite sure what role `isActive` plays now. I don't think it existed when `setFocusedElement` was introduced. 

```js
isActive() {
	return this.editableRef === document.activeElement;
}
```
Okey, so it seems like it has the same limitations as working with `blur` event. It might not work as expected when using popovers or block toolbar controls.

We need to find a good solution because this also causes some confusing bugs in the UI when building custom blocks which contain other components which can take over focus like `PlainText` of any form controls. See #7463 for reference.

Reply:
This might be something we could use [`withFocusOutside`](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/higher-order/with-focus-outside) for, which handles both cases of:

- `blur` is actually a focus change _within_ the element
- _within_ the element might not _within_ the same DOM tree (`withFocusOutside` uses React's virtual bubbling)

Reply:
It works perfectly fine with popover integration for `Link` format control. However it doesn't work properly with mouse interaction in the block toolbar because this slot is rendered outside of the `RichText` component.